internal/registryclient/registryclient.go

```diff
@@ -157,13 +159,17 @@ func (r *GrpcServerRegistry) register() (error, uint32) {
 			Str("IP or Hostname", hostName).
 			Msg("No valid Hostname given. Should be an IP or DNS name.")
 	}
-	portNumber, _ := strconv.Atoi(portNumberString)
 	if err != nil {
 		log.Warn().Err(err).Msg("Could parse GRPC server address")
```

Comment:
Just found this typo. Maybe we can fix it together with the other changes and move the entire error handling block directly after the `net.SplitHostPort(...)`?

```suggestion
		log.Warn().Err(err).Msg("Could not parse GRPC server address")
```

Reply:
Took the comment.